### PR TITLE
layer/getOrphan: improve fs scan loop

### PR DESCRIPTION
**Pull request by vikramhh**
Implements code review comments in https://github.com/moby/moby/pull/39748
closes https://github.com/moby/moby/pull/39748

Signed-off-by: Vikram bir Singh <vikrambir.singh@docker.com>

**- What I did**
Reimplemented iteration over fileInfos in function getOrphan to:

1. Reduce complexity due to nested if blocks by using early return/continue.
2. Improve logging

**- How I did it**
Replaced existing implementation with the suggested one.

**- How to verify it**
Unit test for orphan layers and verifying https://github.com/moby/moby/pull/39193 is not regressed.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vikramhh** [comment]
@olljanat @thaJeztah PTAL

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
If we want https://github.com/moby/moby/pull/39193 to be backported without these changes, let's make sure that one gets merged first, and then rebase this one

**Review by dmcgowan — APPROVED** [review]
LGTM

File: layer/filestore.go [hunk 1]
```diff
@@ -317,32 +317,38 @@ func (fms *fileMetadataStore) getOrphan() ([]roLayer, error) {
...
-
-					l := &roLayer{
-						chainID: chainID,
-						cacheID: cacheID,
-					}
-					orphanLayers = append(orphanLayers, *l)
+			if !fi.IsDir() || !strings.HasSuffix(fi.Name(), "-removing") {
+				continue
+			}
+			encoded := strings.TrimSuffix(fi.Name(), "-removing")
```
Comment (olljanat) [inline layer/filestore.go]: On this phace `fi.Name()` have value like `<digest>-<random>-removing`. Earlier code did split it to three parts so result was only `<digest>` but as you only trim `-removing` then encoding will have value `<digest>-<random>` so cleanup does not work.

**Review by olljanat — CHANGES_REQUESTED** [review]

File: layer/filestore.go (hunk 1 above)
Comment (vikramhh) [reply layer/filestore.go]: @olljanat - modified. PTAL

**Comment by olljanat** [comment]
LGTM

**Pull request merged**